src/meta/Matches.ts

```diff
@@ -0,0 +1,14 @@
+import { Base } from './Base';
+
+export default class Matches extends Base {
+	/**
+	 * Determine if the target of a particular `Event` matches the virtual DOM key
+	 * @param key The virtual DOM key
+	 * @param event The event object
+	 */
+	public get(key: string, event: Event): boolean {
+		this.requireNode(key);
```

Comment:
I see the point.

> 1) cause an immediate re-render

This wouldn't cause an event to re-fire, so really it doesn't do anything.

> 2) throw an error if the node doesn't still doesn't exist in the next render?

Throwing an error when the node doesn't exist feels like the only real way to handle this, it is the only way to properly inform a mistake was made.